src/core/lwt_pool.ml

```diff
@@ -36,6 +36,8 @@ type 'a t = {
   (* Validate old pool members. *)
   dispose : 'a -> unit Lwt.t;
   (* Dispose of a pool member. *)
+  mutable cleared : bool ref;
```

Comment:
The amount of indirection here is a bit sneaky, and the below syntax

```ocaml
p.cleared := true;
p.cleared <- ref false;
```

was mysterious to me when I first saw it. I'd suggest making this an "immutable" field of type `bool ref ref`.